cookiecutter/hooks.py

```diff
@@ -27,17 +27,21 @@
 def valid_hook(hook_file, hook_name):
     """Determine if a hook file is valid.
 
+    Hook name must be a substring of an element in _HOOKS variable to be valid.
+
+    Hook name must be a substring of an element in _HOOKS variable to be valid.
+
     :param hook_file: The hook file to consider for validity
     :param hook_name: The hook to find
     :return: The hook file validity
     """
     filename = os.path.basename(hook_file)
     basename = os.path.splitext(filename)[0]
-    matching_hook = basename == hook_name
-    supported_hook = basename in _HOOKS
+    matching_hook = hook_name in basename
```

Comment:
I think it should be a requirement that file names start with the hook identifier to avoid unintended execution of files. For example, without this constraint, a file like `utils_for_pre_gen_project.py` could run as a hook unintentionally.
```suggestion
    matching_hook = basename.startswith(hook_name)
```